components/dashboard/src/settings/SSHKeys.tsx

```diff
@@ -0,0 +1,258 @@
+            )}
+            {showDelModal && (
+                <DeleteSSHKeyModal
+                    value={currentDelData!}
+                    onConfirm={loadData}
+                    onClose={() => setShowDelModal(false)}
+                />
+            )}
+            <div className="flex items-start sm:justify-between mb-2">
+                <div>
+                    <h3>SSH Keys</h3>
+                    <h2 className="text-gray-500">Create and manage SSH keys.</h2>
+                </div>
+                {dataList.length !== 0 ? (
+                    <div className="mt-3 flex">
+                        <button onClick={addOne} className="ml-2">
+                            New SSH Key
+                        </button>
+                    </div>
+                ) : null}
+            </div>
+            {dataList.length === 0 ? (
+                <div className="bg-gray-100 dark:bg-gray-800 rounded-xl w-full h-96">
+                    <div className="pt-28 flex flex-col items-center w-112 m-auto">
+                        <h3 className="text-center pb-3 text-gray-500 dark:text-gray-400">No SSH Keys</h3>
+                        <div className="text-center pb-6 text-gray-500">
+                            SSH keys allow you to establish a <b>secure connection</b> between your <b>computer</b> and{" "}
+                            <b>workspaces</b>.
+                        </div>
+                        <button onClick={addOne}>New SSH Key</button>
+                    </div>
+                </div>
+            ) : (
+                <div className="grid grid-cols-1 lg:grid-cols-2 gap-4 mt-4">
+                    {dataList.map((key) => {
+                        return (
+                            <Item solid className="items-start">
+                                <KeyItem sshKey={key}></KeyItem>
+                                <ItemFieldContextMenu
+                                    position="start"
+                                    menuEntries={[
+                                        {
+                                            title: "Delete",
+                                            customFontStyle:
+                                                "text-red-600 dark:text-red-400 hover:text-red-800 dark:hover:text-red-300",
+                                            onClick: () => deleteOne(key),
+                                        },
+                                    ]}
+                                />
+                            </Item>
+                        );
+                    })}
+                </div>
+            )}
+        </PageWithSubMenu>
+    );
+}
+
+function KeyItem(props: { sshKey: UserSSHPublicKeyValue }) {
+    const key = props.sshKey;
+    return (
+        <ItemField className="flex flex-col gap-y box-border overflow-hidden">
+            <p className="truncate text-gray-400 dark:text-gray-600">SHA256:{key.fingerprint}</p>
```

Comment:
**suggestion**: Since we're never going to have space to print the whole value here, shall we trim the text here and show only for example 16 characters or somehow encode this long string?

Reply:
> ... although the first dozen or so characters are basically the same

Could we then trim the first characters and show only the last dozens ones? Makes sens or no?

Reply:
To clarify, the reason of asking this is because the encoded string is sometimes not visible to users in smaller viewports like a regular 13" screen. Although in this case, the information is not quite interesting or helpful, it's generally good for users to be able to see the whole information when possible.

| Zoom 100% | Zoom 90% |
|-|-|
| <img width="1440" alt="Screenshot 2022-06-30 at 1 17 33 PM" src="https://user-images.githubusercontent.com/120486/176654551-55985cf4-2937-4fc5-8998-495da2a3aad2.png"> | <img width="1440" alt="Screenshot 2022-06-30 at 1 17 39 PM" src="https://user-images.githubusercontent.com/120486/176654555-f1b64439-7bc2-4c0c-8b6d-32cac3d81a1e.png"> |
 
However, I see that the string is always (?) visible on the delete confirmation modal so let's leave this as is.

| Delete Key (1) | Delete Key (2) |
|-|-|
| <img width="1440" alt="Screenshot 2022-06-30 at 1 17 52 PM" src="https://user-images.githubusercontent.com/120486/176654787-e40c9295-a7f2-43f1-82fe-58b4b347a087.png"> | <img width="1440" alt="Screenshot 2022-06-30 at 1 18 04 PM" src="https://user-images.githubusercontent.com/120486/176654791-d7f8dc8f-aa22-4fff-941f-22f766fbfa35.png"> |

Thanks, @iQQBot! Resolving this thread. 🤝